ethereum/contracts/ERC20App.sol

```diff
@@ -79,10 +81,7 @@ contract ERC20App is AccessControl {
         );
         channel.submit(msg.sender, call);
 
-        require(
-            IERC20(_token).transferFrom(msg.sender, address(this), _amount),
-            "Contract token allowances insufficient to complete this lock request"
-        );
+        IERC20(_token).safeTransferFrom(msg.sender, address(this), _amount);
```

Comment:
I'm not sure this is relevant for us - this line transfers tokens from other contracts into ours, and so permanent lockup may not make sense? I think as is this may actually break the contract because it does not implement IERC721Receiver, which is a requirement for accepting a safeTransferFrom.

Reply:
On looking around the code for the SafeERC20 token which basically uses the low-level call to perform the function call. Also there is no requirement for any IERC20Receiver while using SafeERC20.sol.

Reply:
OK - having said that, I don't think safeTransferFrom adds any security here in the lock function, so is likely wasted gas. I don't think we should use it in this function.